pkg/analytics/fields_listener.go

```diff
@@ -44,27 +45,37 @@ func (fl *fieldListener) Write(entry zapcore.Entry, fields []zapcore.Field) erro
 	allFields = append(allFields, fl.fields...)
 	allFields = append(allFields, fields...)
 
-	safeLogsStr := logging.SanitizeFields(allFields, fl.client.Hash)
+	message := entry.Level.CapitalString()
+
+	for k, v := range logging.SanitizeFields(allFields, fl.client.Hash) {
+		property := fmt.Sprintf("log.%s", k)
+		fl.client.Properties[property] = v
+		defer func() { fl.client.DeleteProperty(property) }()
```

Comment:
This isn't threadsafe which I believe that needs to be a requirement for a fieldListener since calls from concurrent threads/goroutines are not serialized.